valhalla/sif/costconstraints.h

```diff
@@ -0,0 +1,169 @@
+                              const baldr::GraphTile*,
+                              const uint32_t,
+                              const float,
+                              const bool)>
+      StaticComputeConstraintForEdgeF;
+  static std::function<
+      double(uint32_t, const baldr::GraphId& id, const uint32_t start, const float end, const bool)>
+      StaticComputeConstraintForNodeF;
+
+  /**
+   * Parses constraints from json
+   */
+  static void ParseConstraints(const rapidjson::Value& value, CostingOptions* pbf_costing_options) {
+    // auto speed_types = rapidjson::get_child_optional(value, "/speed_types");
+    auto constraint_costing_json =
+        rapidjson::get_optional<rapidjson::Value::ConstArray>(value, "/constraints");
+    if (constraint_costing_json) {
+      for (const auto& constraint : *constraint_costing_json) {
+        auto c = pbf_costing_options->add_costing_constraints();
+        auto max_inclusive = rapidjson::get_optional<bool>(constraint, "/max_inclusive");
+        auto min_inclusive = rapidjson::get_optional<bool>(constraint, "/min_inclusive");
+        if (max_inclusive)
+          c->set_max_inclusive(max_inclusive.get());
+        if (min_inclusive)
+          c->set_min_inclusive(min_inclusive.get());
+
+        auto maxval = rapidjson::get_optional<double>(constraint, "/max_value");
+        auto minval = rapidjson::get_optional<double>(constraint, "/min_value");
+        if (maxval)
+          c->set_max_value(maxval.get());
+        if (minval)
+          c->set_min_value(minval.get());
+      }
+    }
+  }
+  /**
+   * Loads Constraints from costing options into structure...
+   * @param pbf_costing_options Costing pptions to be loaded...
+   */
+  void LoadConstraints(const CostingOptions& pbf_costing_options) {
+    for (auto option = pbf_costing_options.costing_constraints().begin();
+         option != pbf_costing_options.costing_constraints().end(); ++option)
+      constraints_.push_back(CostingConstraint(*option));
+  }
+  /**
+   * Compute all secondary constraints for the specified directed edge.
+   * @param   cost        Cost where constraint calculations will be stored.
+   * @param   edge        Pointer to a directed edge.
+   * @param   tile        Pointer to the tile which contains the directed edge for speed lookup
+   * @param   start       Time arriving at edge (secs)
+   * @param   end         Time leaving edge (secs)
+   * @param   is_forward  Boolean indicating whether this is a forward or reverse search
+   */
+  virtual void ComputeConstraintsForEdge(Cost& cost,
+                                         const baldr::DirectedEdge* edge,
+                                         const baldr::GraphTile* tile,
+                                         const uint32_t start,
+                                         const float end,
+                                         const bool is_forward) const {
+    auto ComputeConstraintForEdgeF = GetComputeConstraintForEdgeF();
```

Comment:
`...EdgeF` meaning `...EdgeForward`?

Reply:
F is short for Function in this context. The function call takes a boolean at the end to handle forward vs backward. Not a lot of thought went into the naming for that function call - I'll see if I can think of a more descriptive replacement.